### PR TITLE
Added --exclude-networks switch.

### PR DESCRIPTION
**Pull request by ghost**
This switch allows to exclude network from exported networks.
In addition networking resources was updated.

### CODE REVIEW

File: src/ralph/export_to_ng/management/commands/export_model.py [hunk 1]
```diff
@@ -125,6 +141,11 @@ class Command(BaseCommand):
             action="append",
             type="str"
         ),
+        make_option(
+            '--exclude-networks',
+            help='Excludes networks from passed range, eg.: --exclude-networks=<min-ip>,<max-ip>', # noqa
```
Comment (mkurek) [inline src/ralph/export_to_ng/management/commands/export_model.py]: min and max ips are "ok", but IMO it'd be better to pass network with mask here (ex. 10.20.30.00/24).
Btw what if I want to exclude many networks? Consider `action="append"` here

File: src/ralph/export_to_ng/resources.py [hunk 2]
```diff
@@ -430,6 +430,12 @@ def get_queryset(self):
 class NetworkResource(resources.ModelResource):
     kind = fields.Field('kind', column_name='kind')
 
+    def get_queryset(self):
+        queryset = models_network.Network.objects.exclude(
+            min_ip__gte=self.min_ip, max_ip__lte=self.max_ip,
```
Comment (mkurek) [inline src/ralph/export_to_ng/resources.py]: what if `min_ip` and `max_ip` are not defined?

File: src/ralph/export_to_ng/management/commands/export_model.py [hunk 3]
```diff
@@ -109,6 +110,21 @@ def get_resource(model_name):
     return resource()
 
 
+def set_options_on_resource(model, options, model_resource):
+    """Injects command line options to resources"""
+    if model == 'Network' and 'exclude_networks' in options:
+        try:
+            min_ip, max_ip = options['exclude_networks'].split(',')
+        except ValueError:
+            print(
```
Comment (vi4m) [inline src/ralph/export_to_ng/management/commands/export_model.py]: Print errors not on stdout, but stderr.

**Comment by ghost** [comment]
TODO: test if data are importable in NG

**Comment by szok** [comment]
:ok:

File: src/ralph/export_to_ng/management/commands/export_model.py [hunk 4]
```diff
@@ -125,8 +131,33 @@ class Command(BaseCommand):
...
+                    IPv4Network(network)
+                except AddressValueError:
+                    self.stdout.write(
+                        "Value '{}' for '--exclude-network' is not CIDR format".format( # noqa
+                            network
+                        )
+                    )
+                    sys.exit()
+                excluded_networks.append(network)
+        model_resource.excluded_networks = excluded_networks
```
Comment (mkurek) [inline src/ralph/export_to_ng/management/commands/export_model.py]: shouldn't it be set only on Network model?

Comment (ghost) [inline src/ralph/export_to_ng/management/commands/export_model.py]: of course :) thanks

File: src/ralph/export_to_ng/resources.py [hunk 5]
```diff
@@ -429,6 +430,16 @@ def get_queryset(self):
 
 class NetworkResource(resources.ModelResource):
```
Comment (mkurek) [inline src/ralph/export_to_ng/resources.py:477]: I'd add here `excluded_networks = []` property if somehow this resource is called outside of mgmt command

File: src/ralph/export_to_ng/resources.py [hunk 6]
```diff
@@ -30,11 +30,29 @@
...
 logger = logging.getLogger(__name__)
 
 
+def get_data_center_id(network_datacenter):
+    """
+    Gets data center id for `network_datacenter`.
+    """
+    try:
+        data_center_id = DataCenter.objects.get(
+            name__iexact=network_datacenter.name
```
Comment (mkurek) [inline src/ralph/export_to_ng/resources.py:46]: couldn't you use `deprecated_ralph_dc` field (https://github.com/allegro/ralph_assets/blob/develop/src/ralph_assets/models_dc_assets.py#L128)?

File: src/ralph/export_to_ng/resources.py [hunk 7]
```diff
@@ -427,8 +445,61 @@ def get_queryset(self):
         return list(chain(parents, children))
 
 
+class NetworkKindResource(resources.ModelResource):
+    class Meta:
+        fields = ('id', 'name',)
+        model = models_network.NetworkKind
+
+
+class DiscoveryDataCenterResource(resources.ModelResource):
```
Comment (mkurek) [inline src/ralph/export_to_ng/resources.py:454]: shouldn't we skip here DCs which have equivalent in assets? comment is suggesting that, but i don't see any code that is doing that

File: src/ralph/export_to_ng/resources.py (hunk 6 above)
Comment (ghost) [inline src/ralph/export_to_ng/resources.py:46]: It ended up, it points to different datacenter (ralph core device with type='datacenter') which means. It's impossible.

File: src/ralph/export_to_ng/resources.py (hunk 7 above)
Comment (xliiv) [inline src/ralph/export_to_ng/resources.py]: I think this docsting is obsolete.
Something more like this
```
This exports only data centers from:
models_network.DataCenter
```
is correct but in such case there is no point to put it at all, imho.

Comment (ghost) [inline src/ralph/export_to_ng/resources.py:454]: Done

**Comment by szok** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**